File: 389.FindTheDifference/Diff.java
import java.util.*;

class Solution {
    public char findTheDifference(String s, String t) {
        Map<Character, Integer> letters = new HashMap<>();
        for (char c : s.toCharArray()) {
            if (!letters.containsKey(c)) {
                letters.put(c, 0);
            }
            letters.put(c, letters.get(c)+1);
        }

        for (char c : t.toCharArray()) {
            if (!letters.containsKey(c) || letters.get(c) == 0) {
                return c;
            }
            letters.put(c, letters.get(c)-1);
        }
        return ' ';
    }
}

File: 389.FindTheDifference/diff.cc
class Solution {
public:
    char findTheDifference(string s, string t) {
        map<char,int> letters;
        for (char c : s) {
            letters[c]++;
        }

        for (char c : t) {
            if (!letters[c]) {
                return c;
            }
            letters[c]--;
        }

        return 0;
    }
};
